src/Groups/GAPGroups.jl

```diff
@@ -372,9 +372,26 @@ end
 
 Base.:*(x::GAPGroupElem, y::GAPGroupElem) = _prod(x, y)
 
-==(x::GAPGroup, y::GAPGroup) = GapObj(x) == GapObj(y)
+function ==(x::GAPGroup, y::GAPGroup)
+  _check_compatible(x, y; error = false) || throw(ArgumentError("x and y are not compatible"))
```

Comment:
```suggestion
  _check_compatible(x, y; error = true)
```
wouldn't this be easier? (similar below)
I don't see a reason to first suppress an error and then throw basically the same one right afterwards

Reply:
Yes, we do not get a better error message in this case.